controllers/FrontController.php

```diff
@@ -475,14 +485,22 @@ protected static function getArrayOnlyWithApprovedKeys($array, $approvedKeys)
      * @param $extraVars
      * @param $customer
      * @param $orderStatus
+     * @throws \PrestaShopException
      */
     private function createOrUpdateOrder($cart, $extraVars, $customer, $orderStatus)
     {
         // Load order if exists from cart id
         if ($cart->OrderExists() !== false) {
             $order = $this->orderAdapter->getOrderByCartId($cart->id);
             if (\Validate::isLoadedObject($order)) {
-                $this->orderService->updateOrderState($order, $orderStatus, $extraVars);
+                $this->orderService->updateOrderState($order, $orderStatus);
+                $latestOrderPayment = $this->orderPaymentService->getLatestAdyenOrderPayment($order);
```

Comment:
I am not sure about this so it's more like a question for this:

Does it make sense to use the latestAdyenPayment object here or maybe the first one makes more sense?
Since we create a new order for every redirect payment attempt and for the sync payments we only create the order when the payment status is not negative (rejected, cancelled, error) then theoretically we only should have 1 adyen payment object per order right?

If so, then I believe here we should update the first payment object instead of the last one.

Regarding its future proofness could you please check if refunds would create any extra order payment object for the order. If so then it's 100% sure that we should use the first payment object instead of the last one since then it would be another payment which we would update.

Could you please double check these concerns before merging it?

Reply:
I'll check it, but I don't think Refunds create a new `order_payment`. So as mentioned we will probably have only 1 adyen order_payment per order

Reply:
okay, let's create a follow up investigation ticket just to be sure we won't forget it and keep it as is for now 👍

Reply:
First `order_payment` is now retrieved: https://github.com/Adyen/adyen-prestashop/pull/177/commits/57cacf292a72bc99996108e8136230896338f057